src/shared/meson.build

```diff
@@ -270,6 +270,15 @@ target2 = custom_target(
 shared_generated_gperf_headers = [target1, target2]
 shared_sources += shared_generated_gperf_headers
 
+fname = 'ethtool-link-mode.h'
```

Comment:
hmm, things like the fstype logic prefer doing this via `cpp -dM`, i.e. let the compiler spit out a list of suitable defines which it then filters.

why depart from that logic here and do it differently? parsing this via cpp sounds more robust for the future to me...